### PR TITLE
feat: Migrated the method getEtxn to platform

### PR DESCRIPTION
**Pull request by devendrafyle**
### Description
copilot:summary

copilot:poem

### Walkthrough
copilot:walkthrough

## Clickup
https://app.clickup.com/t/86cun4u0t

### CODE REVIEW

File: src/app/core/services/transaction.service.ts [hunk 1]
```diff
@@ -836,6 +836,10 @@ export class TransactionService {
     }
   }
 
+  sourceAccountTypePublicMapping(type: string): string {
```
Comment (Julias0) [inline src/app/core/services/transaction.service.ts:839]: where is this method used?

Comment (devendrafyle) [reply src/app/core/services/transaction.service.ts:839]: in key mapping, for `transformRawExpense` method

File: src/app/core/services/transaction.service.ts [hunk 2]
```diff
@@ -949,6 +953,91 @@ export class TransactionService {
...
+      tx_policy_flag: expense.is_policy_flagged,
+      tx_extracted_data: expense.extracted_data
+        ? {
+            vendor: expense.extracted_data?.vendor_name,
+            currency: expense.extracted_data?.currency,
+            amount: expense.extracted_data?.amount,
+            date: expense.extracted_data?.date,
+          }
+        : null,
+      tx_matched_corporate_card_transactions: Array.isArray(expense.matched_corporate_card_transactions)
```
Comment (suyashpatil78) [inline src/app/core/services/transaction.service.ts:1016]: imho, we should extract this logic to a separate method for more readability.

**Review by suyashpatil78 — APPROVED** [review]
One suggestion! Rest LGTM

**Review by Dimple16 — APPROVED** [review]

**Review by Julias0 — APPROVED** [review]

**Pull request merged**